### PR TITLE
Add new resource ReleaseRpmMapping

### PR DESCRIPTION
**Pull request by simozhan**
Add a new resource for downstream project to get metadata

### CODE REVIEW

File: lib/pdc/v1/release_rpm_mapping.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+module PDC::V1
+  class ReleaseRpmMapping < PDC::Base
```
Comment (sthaha) [inline lib/pdc/v1/release_rpm_mapping.rb]: The class needs to be tested. See spec/pdc/v1/ directory

Comment (simozhan) [inline lib/pdc/v1/release_rpm_mapping.rb]: Ack. Added it

File: examples/pdc_resource_tests.rb [hunk 2]
```diff
@@ -171,3 +171,25 @@ def token
     end
   end
 end
+
+# using prod server to test PDC::V1::ReleaseRpmMapping
+pdc = server[:prod]
+
+PDC.configure do |config|
+  config.site = pdc[:site]
```
Comment (ycheng-aa) [inline examples/pdc_resource_tests.rb]: Identical code found in pdc_resource_tests.rb:35, pdc_resource_tests.rb:179 (mass = 104)

File: Gemfile [hunk 3]
```diff
@@ -31,5 +31,11 @@ group :test do
   gem 'timecop'
 end
 
+group :development, :test do
+  # Pronto
+  gem 'pronto'
```
Comment (ycheng-aa) [inline Gemfile]: Bundler/DuplicatedGem: Gem `pronto` requirements already given on line 24 of the Gemfile.

File: Gemfile [hunk 4]
```diff
@@ -31,5 +31,11 @@ group :test do
...
+  gem 'pronto-rubocop', require: false
```
Comment (ycheng-aa) [inline Gemfile]: Bundler/DuplicatedGem: Gem `pronto-rubocop` requirements already given on line 26 of the Gemfile.

File: lib/pdc/resource/per_thread_registry.rb [hunk 5]
```diff
@@ -38,17 +38,19 @@ module PerThreadRegistry
...
     protected
-      def method_missing(name, *args, &block) # :nodoc:
-        # Caches the method definition as a singleton method of the receiver.
-        #
-        # By letting #delegate handle it, we avoid an enclosure that'll capture args.
-        singleton_class.delegate name, to: :instance
-        send(name, *args, &block)
-      end
+
+    def method_missing(name, *args, &block) # :nodoc:
```
Comment (ycheng-aa) [inline lib/pdc/resource/per_thread_registry.rb]: Style/MethodMissing: When using `method_missing`, define `respond_to_missing?` and fall back on `super`. (https://github.com/bbatsov/ruby-style-guide#no-method-missing)

File: lib/pdc/resource/relation.rb [hunk 6]
```diff
@@ -63,19 +63,20 @@ def count
 
     private
 
-      def method_missing(name, *args, &block)
-        # pass anything that relation doesn't know to the klass
-        super unless klass.respond_to? name
+    def method_missing(name, *args, &block)
```
Comment (ycheng-aa) [inline lib/pdc/resource/relation.rb]: Style/MethodMissing: When using `method_missing`, define `respond_to_missing?`. (https://github.com/bbatsov/ruby-style-guide#no-method-missing)

File: pdc.gemspec [hunk 7]
```diff
@@ -3,6 +3,7 @@ lib = File.expand_path('../lib', __FILE__)
 $LOAD_PATH.unshift(lib) unless $LOAD_PATH.include?(lib)
 require 'pdc/version'
 
+# afadsfdasf asdlfjdasljf lakjfldsaj l;sadjfl;asdjf lajsdfldsjal;fj asl;dkjfladsjf ;ljasf;ljsda;lfjdsal;sl;jslj;lj;lsjfla l;kdflksdj
```
Comment (ycheng-aa) [inline pdc.gemspec]: Metrics/LineLength: Line is too long. [132/100] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)

File: Gemfile [hunk 8]
```diff
@@ -1,34 +1,40 @@
...
-  gem 'vcr'
+  gem 'simplecov'
   gem 'timecop'
+  gem 'vcr'
+  gem 'webmock', '~> 1.18.0'
+end
+
+group :development, :test do
+  # Pronto
+  gem 'pronto'
```
Comment (ycheng-aa) [inline Gemfile]: Bundler/DuplicatedGem: Gem `pronto` requirements already given on line 24 of the Gemfile.

File: Gemfile [hunk 9]
```diff
@@ -1,34 +1,40 @@
...
+  gem 'pronto-rubocop', require: false
```
Comment (ycheng-aa) [inline Gemfile]: Bundler/DuplicatedGem: Gem `pronto-rubocop` requirements already given on line 26 of the Gemfile.

**Pull request merged**